app/Http/Controllers/DashboardController.php

```diff
@@ -69,14 +76,84 @@ public function getFriends()
+        $notification->category = NotificationCategoryEnum::EXPENSE;
+        $notification->type = NotificationTypeEnum::REQUEST;
+        $notification->parameters = $sharedExpense->amount_owed;
+        $notification->reference_id = $sharedExpense->id;
+        $notification->save();
+    }
+
+    /**
+     * Get notifications where user is the recipient of the friend request.
+     * @return an array of arrays. Each sub array contains 1. Notification object, 2. Notification message (string)
+     */
+    public function getFriendNotifications()
+    {
+        $requests = DB::table('notifications')
+            ->where('recipient_username', Auth::user()->username)
+            ->where('category', 2) //category 2 is for just Friend notifications
+            ->where('type', 1) //Just request notifications
+            ->orderBy('is_read', 'ASC')
+            ->get();
+        $notifications = array();
+        foreach($requests as $request)
+        {
+            $n = new FriendNotification($request->recipient_username,
+                $request->sender_username,
+                $request->category, $request->type,
+                $request->parameters, $request->reference_id);
+
+            $notification = array($n, $n->messageForNotification($n));
+            array_push($notifications, $notification);
+        }
+        return $notifications;
+    }
+
+    /**
+     * Get expense request notifications where the user is the recipient.
+     * @return array holding subarrays of 1. Notification object, 2. Notification message as string
+     */
+    public function getExpenseNotifications()
+    {
+        $requests = DB::table('notifications')
+        ->where('recipient_username', Auth::user()->username)
+        ->where('category', 1) //only expense notifications
+        ->where('type', 1) //only requests -- uncomment for all types
+        ->orderBy('is_read', 'ASC')
+        ->get();
+        $notifications = array();
+        foreach($requests as $request) {
+            $n = new ExpenseNotification($request->recipient_username,
+                $request->sender_username,
+                $request->category,
+                $request->type,
+                $request->parameters,
+                $request->reference_id);
+            $notification = array($n, $n->messageForNotification($n));
+            array_push($notifications, $notification);
+        }
+        return $notifications;
+    }
+
+    /**
      * Show the application dashboard.
      *
      * @return \Illuminate\Http\Response
      */
     public function index()
     {
         $expenses = $this->getExpenses();
-        //$friends = getFriends();
-        return view('dashboard', compact('expenses'));
+        $friendNotifications = $this->getFriendNotifications();
+        $expenseNotifications = $this->getExpenseNotifications();
+        return view('dashboard', compact('expenses', 'friendNotifications', 'expenseNotifications'));
```

Comment:
The way the notifications are designed right now would break the other pages. Because this means the notifications would have to be passed into each views because the sidebar is trying to access a $notifications variable.

I suggest that we retrieve the notifications by Ajax request.